rclcpp/include/rclcpp/service.hpp

```diff
@@ -163,26 +203,18 @@ class Service : public ServiceBase
     if (!service_name) {
       throw std::runtime_error("failed to get service name");
     }
-    service_handle_ = service_handle;
     service_name_ = std::string(service_name);
-    owns_rcl_handle_ = false;
+
+    // In this case, rcl owns the service handle memory
+    service_handle_ = std::shared_ptr<rcl_service_t>(new rcl_service_t);
+    service_handle_->impl = service_handle->impl;
   }
 
+
```

Comment:
nitpick: minimize vertical whitespace (for reference, this is part of the Google Style guide which our style is based on, see: https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace)